### ISSUE TITLE
[Question] Plotting training and validation loss? 

### ISSUE THREAD

**Issue by jsolson4** [manujosephv/pytorch_tabular#13]
This package has been super helpful, I appreciate all the effort this must have taken. 

I've been wanting to better understand the training process and effectiveness of additional epochs. Is there a way to print or plot training and validation loss? Or if anyone has developed something similar, it would be cool to see. I am still trying to figure out access to components of the model training like that. 

I think opening the discussion would be super helpful as well :). I think you (as the owner) can just hit the selection and it will open.

**Comment by manujosephv** [comment]
There are two ways you can monitor the training loss/ validation loss. 
1. TensorBoard
2. Weights and Biases

By default, all the training metrics are logged into TensorBoard and you can monitor them by starting the Tensorboard server. Another option(one I prefer) is to use Weights and Biases, which is more feature rich than TensorBoard.

In addition to plotting Training Losses, you can plot any metric that you provide, gradients of all the layers through epochs, the norm of the gradients, the parameter distribution across epochs, and the logits across epochs. All of this just by using a few parameters in the config.

It is mentioned in this [page in the Docs](https://pytorch-tabular.readthedocs.io/en/latest/experiment_tracking/)

Hope this is what you are looking for.

**Comment by manujosephv** [comment]
@jsolson4 Did this answer your question? or are you looking for something different?

**Comment by jsolson4** [comment]
It did answer my question, thank you. Weights and biases looks like an interesting solution, although I am unable to use it for my current project. I've been running a category embedding regression model in DataBricks. I played around with TensorBoard for a bit, but wasn't quite able to get it working. I think this primarily comes down to understanding which paths DataBricks uses and when. 

I've found that my models have been training in a relatively low number of epochs, so I have been relying on early stopping as a work-around to a visual check of training losses, as I needed to get an experiment running. The experiment is focused on finding the best trade-off between input parameters, while holding a vanilla DNN and it's hyperparameters constant, so I don't require perfect training/validation loss patterns. I plan to come back and work on implementing TensorBoard when I get some time in the near future. When that happens, I will let you know if I have any further questions. Thanks for the follow-up, I appreciate it.

Maybe you could explain the learning rate setting within the modelConfig() and how that relates to the optimizer configuration ( I haven't been able to find in the documentation). I would expect the optimizer config (when set) to determine the best learning rate.  However, it seems that the learning rate argument is required regardless. What's happening under the hood, is the learning rate used as a starting point and subsequent LRs are determined via the optimizer of choice? Why the learning rate argument is required in the presence of optimizer config?

**Comment by manujosephv** [comment]
@jsolson4  Sorry, I totally missed the second part of the question.
The learning rate setting within the `ModelConfig` is what is used while initializing the optimizers. `OptimizerConfig` doesn't have a parameter called learning_rate. If `auto_lr_find` is turned on, the learning rate you give in `ModelConfig` is overwritten with the new LR. If `auto_lr_find` is False, then the learning rate you gave in the ModelConfig is the learning rate that is going to be used for training.

And regarding the first part, if TensorBoard is chosen, PyTorch Tabular, by default, saves the files in the CWD under the folder name of  `project_name`. If you can't get Tensorboard server to work, I'm sure you can access the files and parse them yourselves.

**Comment by manujosephv** [comment]
Closing the issue. Feel Free to reopen if it you need more clarifications.
